tool/gravity/cli/config.go

```diff
@@ -1352,6 +1357,33 @@ func checkLocalAddr(addr string) error {
 		addr, strings.Join(availableAddrs, ", "))
 }
 
+// checkOverlayNet verifies that the host network address is not in the range of the serviceCIDR or podCIDR
+func checkOverlayNet(advertiseAddr string, serviceCIDR string, podCIDR string) error {
```

Comment:
I think that in addition to checking the advertise address itself we also need to check other network interfaces on the node in case of the multi-homed node? @knisbet What do you think?

Reply:
I think this gets into rabbit hole territory in the context of detecting conflicts via cluster configuration on a presumably running cluster. For overlay network conflicts it's probably feasible, since the overlay network doesn't get created on any virtual interfaces in the host network namespace. Although there might be alternate implementations out their that would show an interface in the host NS we don't expect. 

Pod cidr would be a whole lot harder, as on a running cluster there may be interfaces assigned within the range on the host (think the flannel bridge). While we can probably control for our own interfaces, this would conflict if anyone tries to use their own CNI plugin. 

So my 2 cents would be to go with the minimal implementation for now. I think to properly check all network interfaces, would require some attention to detail, to only filter to say non virtual / non internal interfaces.